111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,232 @@
+For Buildah, we will check only the `additionalOptions` field.
+In case that user will specify `additionalKanikoOptions` with Buildah enabled feature gate, warning about deprecation and about ignored field will be added to the `.status` section again.
+
+Following table shows what was described above and how these two fields will be checked and handled:
+
+| `additionalKanikoOptions` used | `additionalOptions` used | `UseConnectBuildWithBuildah` in Beta | Buildah used | Warning about deprecation | Warning about ignored field | Used field                |
+|--------------------------------|--------------------------|--------------------------------------|--------------|---------------------------|-----------------------------|---------------------------|
+| ☑️                             | ❌                        | ❌                                    | ❌            | ☑️                        | ❌                           | `additionalKanikoOptions` |
+| ☑️                             | ❌                        | ☑️                                   | ❌            | ☑️                        | ☑️                          | none                      |
+| ☑️                             | ❌                        | ❌️                                   | ☑️           | ☑️                        | ☑️                          | none                      |
+| ☑️                             | ❌                        | ☑️                                   | ☑️           | ☑️                        | ☑️                          | none                      |
+| ☑️                             | ☑️                       | ❌                                    | ❌            | ☑️                        | ☑️                          | `additionalOptions`       |
+| ☑️                             | ☑️                       | ☑️                                   | ❌            | ☑️                        | ☑️                          | `additionalOptions`       |
+
+As in case of Kaniko, Buildah will have also some allowed options that can be specified during the `push` phase:
+- `--authfile` - path to the authentication file (usually created using the `buildah login`).
+- `--cert-dir` - path to directory with certificates that should be used for connecting to registries.
+- `--creds` - the `[username[:password]]` to use to authenticate with the registry if required.
+- `--retry` - number of times to retry in case of failure during image pull (the base one).
+- `--retry-delay` - duration between retry attempts.
+- `--tls-verify` - skip TLS verification for insecure registries.
+
+If the user-specified Buildah options will contain forbidden options (or not known), user will be notified by message inside the `.status` section of `KafkaConnect` resource, the `InvalidResourceException` will be thrown (logged inside the operator log) and the build will fail.
```

Comment:
So, you should probably just say that reconciliation will fail with `InvalidResourceException`. Or do you plan to add some special messages tot he status?